detectors/opentelemetry_resource_detector_aws/src/otel_resource_aws_ecs.erl

```diff
@@ -0,0 +1,140 @@
+-module(otel_resource_aws_ecs).
+
+-behavior(otel_resource_detector).
+
+-export([get_resource/1]).
+
+-include_lib("kernel/include/logger.hrl").
+-include_lib("opentelemetry_semantic_conventions/include/incubating/attributes/cloud_attributes.hrl").
+-include_lib("opentelemetry_semantic_conventions/include/incubating/attributes/container_attributes.hrl").
+-include_lib("opentelemetry_semantic_conventions/include/incubating/attributes/aws_attributes.hrl").
+
+-define(OS_ENV, "ECS_CONTAINER_METADATA_URI_V4").
+-define(DEFAULT_CGROUP_PATH, "/proc/self/cgroup").
+
+%% @private
+-spec get_resource(list()) -> otel_resource:t() | no_return().
+get_resource(Config) ->
+    otel_resource:create(fetch(Config)).
+
+fetch(Config) ->
+    case os:getenv(?OS_ENV) of
+        false ->
+          erlang:error('Not running in ECS environment, ECS_CONTAINER_METADATA_URI_V4 not set.');
+        _MetadataV4URL ->
+          [
+            {?CLOUD_PROVIDER, atom_to_list(?CLOUD_PROVIDER_VALUES_AWS)},
+            {?CLOUD_PLATFORM, atom_to_list(?CLOUD_PLATFORM_VALUES_AWS_ECS)} |
+            aws_ecs_attributes(Config)
+          ]
+    end.
+
+aws_ecs_attributes(Config) ->
+  {ok, Hostname} = inet:gethostname(),
+
+  ContainerIDAttributes =
+    case fetch_container_id_attributes(Config) of
+      {ok, ContainerIDAttributesResult} ->
+        ContainerIDAttributesResult;
+      {error, ContainerIDAttributesError} ->
+        ?LOG_ERROR("Failed to fetch Container ID attributes: ~s", [ContainerIDAttributesError]),
+        []
+    end,
+  
+  MetadataV4Attributes =
+    case fetch_metadata_v4_attributes() of
+      {ok, MetadataV4AttributesResult} ->
+        MetadataV4AttributesResult;
+      {error, MetadataV4AttributesError} ->
+        ?LOG_ERROR("Failed to fetch metadata attributes: ~s", [MetadataV4AttributesError]),
+        []
+    end,
+  
+  [{?CONTAINER_NAME, Hostname}] ++ ContainerIDAttributes ++ MetadataV4Attributes.
+
+fetch_container_id_attributes(Config) ->
+  Path = proplists:get_value(cgroup_path, Config, ?DEFAULT_CGROUP_PATH),
+  
+  case file:read_file(Path) of
+      {ok, Data} ->
+          Lines = binary:split(Data, <<"\n">>, [global]),
+          ContainerId = lists:foldl(fun
```

Comment:
Can you add a comment for how this foldl is supposed to work? It seems to loop through Lines from a file until one is found that is greater than 64 bytes and then take the last 64 bytes? It then continues to iterate through the list but doing nothing since its already found the thing it wanted? You could use `lists:search` and then extract the part of the Line you want, if that is the case.

Why is that the line with the container id?

Reply:
It's a pattern laid down in other instrumentation SDKs with go and js:

https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/detectors/aws/ecs/ecs.go#L250-L254
https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/detectors/node/opentelemetry-resource-detector-aws/src/detectors/AwsEcsDetector.ts#L157-L177

I don't know exactly why it's done this way. We just want to find the container id which is a 64 byte hex binary, but personally I would have pattern matched on `/docker` instead, but I don't know how cgroup file may look. So I've just kept it like the others.